en/docs/learn/api-security/oauth2/access-token-types/jwt-tokens.md

```diff
@@ -192,4 +192,65 @@ Follow the instructions below to import the public certificate into the client t
      keytool -import -trustcacerts -keystore client-truststore.jks -alias gateway_certificate_alias -file wso2.crt
      ```
 
-5. Enter `wso2carbon` as the default password of the trust store when prompted.
\ No newline at end of file
+5. Enter `wso2carbon` as the default password of the trust store when prompted.
+
+## Validating JWT generated From external Oauth Provider
+
+JWT generated from an external Oauth provider can validate through Gateway as per the following diagram.
+
+### Signature Validation flow.
+
+![]({{base_path}}/assets/img/learn/external-jwt-signature-validation.png)
+
+In order to do the signature validation of jwt, can use one of the following steps.
+
+- Import the public certificate into client-truststore.jks located under `<API-M_HOME>/repository/resources/security/` by using **kid** value as alias of key.
+
+- Use JWK endpoint of Oauth Provider.
+
+ 1. Navigate to the `deployment.toml` under the `<API-M_HOME>/repository/conf` folder.
+ 2. Add following content under the `[[apim.jwt.issuer]]` for issuer.
+  ```
+  [[apim.jwt.issuer]]
+  name = "<issuer value of jwt>"
+  [apim.jwt.issuer.jwks]
+  url = "<jwks endpoint of oauth provider>"
+  ```
+
+### Transform JWT Claims
+
+By default WSO2 API Manager can convert the JWT claims into desired claims by configuring the claim mapping as per the following steps.
+
+1. Navigate to the `deployment.toml` under the `<API-M_HOME>/repository/conf` folder.
+2. Add the claim mapping as following under the `[[apim.jwt.issuer]]`.
+  ```
+  [[apim.jwt.issuer.claim_mapping]]
+  remote_claim = "<remote claim value>"
+  local_claim = "<local claim value to map>"
+  ....
+  ```
+By Default Claim mapping done in as one to one mapping, If claim mapping needs to be done in advance behaviour, Users can write a JWT transformer that can configure to specific token issuer.
+
```

Comment:
```suggestion
By default, the claim mapping is done in a one-to-one manner. But, if the mapping needs to be done in an advanced manner, the users can write custom JWT transformers which can be attached to each token issuer.
```

Reply:
@tharindu1st we need to talk more about or link for more details on how to extend this.